core/src/mindustry/ai/BlockIndexer.java

```diff
@@ -415,7 +414,7 @@ public Tile findClosestOre(float xp, float yp, Item item){
                     var arr = ores[item.id][qx][qy];
                     if(arr != null && arr.size > 0){
                         Tile tile = world.tile(arr.first());
-                        if(tile.block() == Blocks.air){
+                        if((floor && tile.block() == Blocks.air) || (wall && tile.wallDrop() != null)){
```

Comment:
Consider this: A unit may only be able to mine floor ores but the first ore in the chunk may be a wall ore, this function only checks the first ore per chunk so it would fail to find a floor ore in this chunk.

Reply:
I don't really know much about how ores are indexed, so I don't know what the best way to handle this is.
Maybe just make chunks not contains wall and floor ores? Separate wall and floor into different chunks?

Reply:
Yeah, when I implemented this this morning (with my admittedly terrible solution) I split them into their own lists

Reply:
So how would I implement it?

Reply:
[My last two commits](https://github.com/mindustry-antigrief/mindustry-client/commits/v7/core/src/mindustry/ai/BlockIndexer.java) to foo's that have changed this file may be a decent starting point but the code for that is still kind of scuffed